Desktop: New design for "New note" and "New todo" buttons

julien-me opened the pull request:
This PR implements a new design to create new todos and new notes in an attempt to make these two actions unambiguous and make these two buttons stand out more among other buttons.

This new design is based on a new UI proposal that was once considered: https://global.discourse-cdn.com/standard14/uploads/cozic/original/2X/2/26c255d41e5a128eb8a5579474e9627768b518db.png

See below screenshots of the result (with light and dark themes) followed by a possible alternative design.

#### New design
![1](https://user-images.githubusercontent.com/32807437/219040212-139ce5e6-1785-4951-9d4a-636d546fd0e7.png)
![1-dark](https://user-images.githubusercontent.com/32807437/219040222-43ddb2ee-285e-4654-9e0a-aa5cb95299ed.png)
#### Possible alternative
![2](https://user-images.githubusercontent.com/32807437/219040227-14c9eea9-a251-4d53-a543-42bf6d4220b9.png)

laurent22 commented on packages/app-desktop/gui/NoteListControls/NoteListControls.tsx:
```diff
@@ -116,7 +126,32 @@ function NoteListControls(props: Props) {
+				<StyleNewTodoButton
+					className="new-todo-button"
+					tooltip={CommandService.instance().label('newTodo')}
+					title="+ New to do"
```
that should be translated, so something like `_('+ %s', 'New to-do')`. And in the UI we generally use the term "to-do" with a dash

julien-me replied:
Fixed.

laurent22 commented on packages/app-desktop/gui/NoteListControls/NoteListControls.tsx:
```diff
@@ -116,7 +126,32 @@ function NoteListControls(props: Props) {
+				<StyledNewNoteButton
+					className="new-note-button"
+					tooltip={CommandService.instance().label('newNote')}
+					title="+ New note"
```
Should also be translated

julien-me replied:
Fixed.

laurent22 commented:
The first design would work, but we need the same padding above and below the group of controls. So there's too much padding below the search input.

And now that I look at it, I feel the "New note" button should be to the left, since it's the most important one

tomasz1986 commented:
> This new design is based on a new UI proposal that was once considered: [global.discourse-cdn.com/standard14/uploads/cozic/original/2X/2/26c255d41e5a128eb8a5579474e9627768b518db.png](https://global.discourse-cdn.com/standard14/uploads/cozic/original/2X/2/26c255d41e5a128eb8a5579474e9627768b518db.png)

The pluses look kind of small when compared to the old screenshot. Why is it so? Is it possible to make them larger? Also, what happens if you shrink the note list horizontally even more? Do the buttons still fit then?

julien-me commented:
>The first design would work, but we need the same padding above and below the group of controls. So there's too much padding below the search input.

~~The extra padding must have been because of flexbox's available space. Is it correct now?~~ Edit: Still needs to be fixed.
![padding-fixed](https://user-images.githubusercontent.com/32807437/219058174-eeca0b95-22b1-4b4c-a84c-3b1ebe2e1838.png)


>And now that I look at it, I feel the "New note" button should be to the left, since it's the most important one

I've updated the code and swapped them.

julien-me commented:
>The pluses look kind of small when compared to the old screenshot. Why is it so? Is it possible to make them larger? 

I'll try making it larger.

> Also, what happens if you shrink the note list horizontally even more? Do the buttons still fit then?

It shrinks as below.
![shrink-buttons](https://user-images.githubusercontent.com/32807437/219058881-51c82cce-ec2d-4861-b332-b154f19d3f07.png)

laurent22 commented:
For the "+" I think it should be an icon? Could you find one in Font Awesome please? I think the current button component supports icons

Daeraxa commented:
Does this PR take into account localisation for the labels?
Edit - ignore that it looks like it does.

Daeraxa commented:
> For the "+" I think it should be an icon? Could you find one in Font Awesome please? I think the current button component supports icons

`fas fa-plus` is already being used for the new notebook icon.

Is there any merit in extending this same metaphor to the notebook list so it also has a captioned button? Personally I like the look of it as it is but it means there are essentially two separate metaphors for similar actions.

laurent22 commented:
> Is there any merit in extending this same metaphor to the notebook list so it also has a captioned button? Personally I like the look of it as it is but it means there are essentially two separate metaphors for similar actions.

I'd rather not add a button there, if only to save space. Perhaps that button should simply be removed, as Julien mentioned in another thread, since it's not as frequently used, but that's something we'll see some other times.

julien-me commented:
>The first design would work, but we need the same padding above and below the group of controls. So there's too much padding below the search input.

Fixed. I've set `height` for the group of controls to `auto`  and removed some extra code computing height of the items list. Since `NoteListControls` and `NoteList` are in a flexbox I don't think it's necessary to hardcode the height in `px`.

laurent22 commented:
Thanks Julien. Also could you try to put "New note", "New to-do" and the sorting buttons on the same line, with search below please? The way things are aligned at the moment doesn't look very good (buttons are centered, but search and filters are right-aligned). If you could post a screenshot of the result so that we can compare that would be great. Or feel free to use something like Figma if it's easier, since it's just to see how it looks

tomasz1986 commented:
> > Also, what happens if you shrink the note list horizontally even more? Do the buttons still fit then?
> 
> It shrinks as below. ![shrink-buttons](https://user-images.githubusercontent.com/32807437/219058881-51c82cce-ec2d-4861-b332-b154f19d3f07.png)

Thank you. Just one more question, what about if you widen the note list? Currently, everything stays nicely in a single line, e.g. when using this kind of layout:

![image](https://user-images.githubusercontent.com/5626656/219345429-acf38059-188f-4f2d-b3e2-9f61c2efbd10.png)

Just wondering whether it could be possible to do the same with the new buttons (when there is enough space, of course). Otherwise they are probably going to end up stretched to the extreme… Is this right?

laurent22 commented:
> It shrinks as below. ![shrink-buttons](https://user-images.githubusercontent.com/32807437/219058881-51c82cce-ec2d-4861-b332-b154f19d3f07.png)

By the way, the buttons should not wrap like this - the text and icon should stay on one line. Maybe you can make it responsive and remove the "+" icon below a certain size? It might also mean that the min width for note list needs to be increased.

julien-me commented:
>Thanks Julien. Also could you try to put "New note", "New to-do" and the sorting buttons on the same line, with search below please? The way things are aligned at the moment doesn't look very good (buttons are centered, but search and filters are right-aligned). If you could post a screenshot of the result so that we can compare that would be great. Or feel free to use something like Figma if it's easier, since it's just to see how it looks

Like this?

![same-line](https://user-images.githubusercontent.com/32807437/219344845-43a76cfa-3b24-4a40-881f-b1cf8aa240e4.png)

julien-me commented:
> For the "+" I think it should be an icon? Could you find one in Font Awesome please? I think the current button component supports icons

With an icon
![fas-icon](https://user-images.githubusercontent.com/32807437/219357067-0c02555d-8051-4562-92b6-eb9ce41054f5.png)

julien-me commented:
> Thank you. Just one more question, what about if you widen the note list? Currently, everything stays nicely in a single line, e.g. when using this kind of layout:

This layout?
![top-layout](https://user-images.githubusercontent.com/32807437/219357793-18e97a75-6449-468d-a3ce-ead39fcefbed.png)

> Just wondering whether it could be possible to do the same with the new buttons (when there is enough space, of course). Otherwise they are probably going to end up stretched to the extreme… Is this right?

Yes they are streched. What would you suggest? Having them on the right of the search bar?

laurent22 commented:
> With an icon

Is it possible to make it look like the original design? The + now is too big

laurent22 commented:
> Like this?

Yes. Hmm, but not sure if it's better now. Maybe we should have some poll in the forum with the two options and ask which one looks better

julien-me commented:
> Yes. Hmm, but not sure if it's better now. Maybe we should have some poll in the forum with the two options and ask which one looks better

Personally I have no opinion because I like them both. Maybe the advantage of having the sorting buttons with the search bar is that it starts to shrink later. 

Also having them on the same line maybe "steals" the focus of the two other and more important (to me) `New` buttons.

tomasz1986 commented:
> This layout? ![top-layout](https://user-images.githubusercontent.com/32807437/219357793-18e97a75-6449-468d-a3ce-ead39fcefbed.png)
> 
> > Just wondering whether it could be possible to do the same with the new buttons (when there is enough space, of course). Otherwise they are probably going to end up stretched to the extreme… Is this right?
> 
> Yes they are streched. What would you suggest? Having them on the right of the search bar?

Yeah, I think it would be better, if possible without too much hackery, and only when there is enough space, of course. The stretched search bar isn't a problem, as it is still useful (so that you can input and see more text), but there is no use in having the buttons stretched so much…

For the record, I use this layout, because I usually use Joplin on smaller/non-wide screens and there is no space for the default three-column layout there.

julien-me commented:
> By the way, the buttons should not wrap like this - the text and icon should stay on one line. 

Text and icon are now on one line. This is the current behavior.

![shrink-note-controls](https://user-images.githubusercontent.com/32807437/219601801-b6ed9a9a-97e1-43f4-9c97-8a6bbacf163f.gif)


>Maybe you can make it responsive and remove the "+" icon below a certain size? It might also mean that the min width for note list needs to be increased.

Removing `+` when the size changes could be done with [container queries](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Container_Queries) but `styled-components` doesn't support them at the moment.

julien-me commented:
I changed the size of the `+` to match the proportion of the size of the text.

![plus-size](https://user-images.githubusercontent.com/32807437/219602836-96d127fa-05e7-4697-a805-bf6344c4dc8e.png)

laurent22 commented:
Thanks @julien-me, I think the way it works when resizing is fine. The "New note" button remains visible the longest, and that's the most important button. The "+" icons look good too so let's merge
